common/scripts/babel-conditional-preprocess.js

```diff
@@ -0,0 +1,80 @@
+Object.defineProperty(exports, '__esModule', { value: true });
+
+const helper = require('@babel/helper-plugin-utils');
+const t = require('@babel/types');
+exports.default = helper.declare((_api, opts) => {
+  const {
+    annotations
+  } = opts;
+
+  function Handle(path) {
+    let { node } = path;
+    let removed = false;
+
+    for (const annotation of annotations) {
+      const {
+        match
+      } = annotation;
+
+      if (node.leadingComments && node.leadingComments.length > 0) {
+        for (const comment of node.leadingComments) {
+          if (comment.value.includes(match)) {
+            removed = true;
+          }
+          if(removed) {
+            comment.ignore = true;
+          }
+        }
+      }
+
+      if (removed) {
```

Comment:
Either:

* Add comments explaining what is happening here ++ only remove if `path.parentPath` has only one child.

Or:

* Just document that we don't support JSX expressions. We'd have to create private Component instead.